grpc/disruption_listener_service.go

```diff
@@ -0,0 +1,126 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	pb "github.com/DataDog/chaos-controller/grpc/disruption_listener"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// DisruptionListener is a gRPC Service that can disrupt endpoints of a gRPC server
+type DisruptionListener struct {
+	pb.UnimplementedDisruptionListenerServer
+	Spec GrpcDisruptionSpec
+}
+
+// GrpcDisruptionSpec configures the DisruptionListener to chaos test endpoints of a gRPC server
+type GrpcDisruptionSpec []EndpointAlteration
+
+// EndpointAlteration configures endpoints that the DisruptionListener chaos tests on a gRPC server
+type EndpointAlteration struct {
+	TargetEndpoint   string
+	ErrorToReturn    string
+	OverrideToReturn string
+}
+
+var (
+	errorMap = map[string]codes.Code{
+		"OK":                  codes.OK,
+		"CANCELED":            codes.Canceled,
+		"UNKNOWN":             codes.Unknown,
+		"INVALID_ARGUMENT":    codes.InvalidArgument,
+		"DEADLINE_EXCEEDED":   codes.DeadlineExceeded,
+		"NOT_FOUND":           codes.NotFound,
+		"ALREADY_EXISTS":      codes.AlreadyExists,
+		"PERMISSION_DENIED":   codes.PermissionDenied,
+		"RESOURCE_EXHAUSTED":  codes.ResourceExhausted,
+		"FAILED_PRECONDITION": codes.FailedPrecondition,
+		"ABORTED":             codes.Aborted,
+		"OUT_OF_RANGE":        codes.OutOfRange,
+		"UNIMPLEMENTED":       codes.Unimplemented,
+		"INTERNAL":            codes.Internal,
+		"UNAVAILABLE":         codes.Unavailable,
+		"DATALOSS":            codes.DataLoss,
+		"UNAUTHENTICATED":     codes.Unauthenticated,
+	}
+)
+
+// SendDisruption makes a gRPC call to DisruptionListener
+func (d *DisruptionListener) SendDisruption(ctx context.Context, ds *pb.DisruptionSpec) (*emptypb.Empty, error) {
+	if ds == nil {
+		log.Fatal("Cannot execute SendDisruption when DisruptionSpec is nil")
```

Comment:
I think we should return an error here too. ( in fact, possibly _instead_ rather than too. This log could/should be at Error )